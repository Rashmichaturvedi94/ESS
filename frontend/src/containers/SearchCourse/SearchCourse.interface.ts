export interface SearchCourseProps {
  myProp?: boolean;
  /* Add props here. */
}
