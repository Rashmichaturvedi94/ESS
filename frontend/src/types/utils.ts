export type Nullable<T> = {
  [P in keyof T]: T[P] | null;
};

export type Uri = string;

export type DateString = string;

export type LocalOrRemoteFile = File | Uri;

export type Modify<T, R> = Omit<T, keyof R> & R;
