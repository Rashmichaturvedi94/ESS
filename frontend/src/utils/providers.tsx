import React, { ComponentType, FC } from 'react';
import * as Sentry from '@sentry/react';
import { QueryClientProvider } from 'react-query';
import DateFnsUtils from '@date-io/date-fns';
import { ThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { queryClient } from '../api';
import { theme } from '../styles/theme';


const createProviderHOC = (Provider: ComponentType) => <P extends object>(
  Component: ComponentType<P>,
) => (props: P) => (
  <Provider>
    <Component {...props} />
  </Provider>
);
const FallbackComponent = () => <div>An error has occurred</div>;

export const Providers: FC = ({ children }) => (
  <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
    <StylesProvider injectFirst>
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <QueryClientProvider client={queryClient}>
              {children}
            </QueryClientProvider>
          </MuiPickersUtilsProvider>
        </ThemeProvider>
      </MuiThemeProvider>
    </StylesProvider>
  </Sentry.ErrorBoundary>
);

export const withProviders = createProviderHOC(Providers);
