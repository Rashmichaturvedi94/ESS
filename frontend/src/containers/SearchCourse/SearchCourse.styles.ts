import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { SearchCourseProps } from './SearchCourse.interface';

export const SearchCourse = styled.div<Pick<SearchCourseProps, 'myProp'>>`
  /* Add styles here */
`;

export const GridListTileLink = styled(Link)`
 text-decoration: none;
 color: initial;
 
 &.MuiGridListTile-tile {
 display: flex;
 }
`;