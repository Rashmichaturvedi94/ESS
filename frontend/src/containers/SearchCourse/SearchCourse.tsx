
import React, { FC } from 'react';
import { useHistory, generatePath  } from 'react-router-dom';
import { GridList, Typography, GridListTile } from '@material-ui/core';
import { SearchCourseProps } from './SearchCourse.interface';
import { useCourses } from '../../api';
import { appPaths } from '../../const/paths';
import { Course } from '../../api/api.interface';

export const SearchCourse: FC<SearchCourseProps> = () => {
  const courses = useCourses();
  const history = useHistory();
  const getImage = (course: Course) => typeof course.img === 'string' ? course.img : '';

  return (
    <GridList>
      {courses.data?.map((course) => (
        <GridListTile 
          key={course.id} 
          style={{ height: 258, width: 224, border: '2px solid black', margin: 8}} 
          onClick={() => history.push(generatePath(appPaths.courseDetail, { courseId: course.id }))}>
          <img src={getImage(course)} style={{ height: 196, width:220, overflow: 'hidden' }} alt={course.title}/>
          <Typography>{course.title}</Typography>
          <Typography>{course.description}</Typography>
        </GridListTile>
      ))}
    </GridList>
  );
};
