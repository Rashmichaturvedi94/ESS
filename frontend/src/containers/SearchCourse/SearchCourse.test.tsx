import React from 'react';
import { render } from '../../utils/testUtils';
import { SearchCourse } from './SearchCourse';

describe('components/SearchCourse', () => {
  it('should render', () => {
    const { container } = render(<SearchCourse />);
    expect(container).toMatchSnapshot();
  });
});
