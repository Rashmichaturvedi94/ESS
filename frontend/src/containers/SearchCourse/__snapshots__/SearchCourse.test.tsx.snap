// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SearchCourse should render 1`] = `
<div>
  <ul
    class="MuiGridList-root"
    style="margin: -2px;"
  />
</div>
`;
